src/main/java/com/jsonnetplugin/JsonnetCompletionContributor.java

```diff
@@ -228,31 +227,31 @@ private static JsonnetExpr getField(JsonnetObjinside obj, String name) {
     }
 
     private static JsonnetObjinside[] resolveExpr0ToObj(JsonnetExpr0 expr0, List<JsonnetExpr> visited) {
-        if (expr0.getIfelse() != null){
+        if (expr0.getIfelse() != null) {
             JsonnetIfelse ifelse = expr0.getIfelse();
 
             ArrayList<JsonnetObjinside> output = new ArrayList<>();
             JsonnetObjinside[] lhsRes = resolveExprToObj(ifelse.getExprList().get(1), visited);
-            if (lhsRes != null){
-                for(JsonnetObjinside j: lhsRes) output.add(j);
+            if (lhsRes != null) {
+                output.addAll(Arrays.asList(lhsRes));
             }
-            if (ifelse.getExprList().size() == 3){
+            if (ifelse.getExprList().size() == 3) {
                 JsonnetObjinside[] rhsRes = resolveExprToObj(ifelse.getExprList().get(2), visited);
-                if (rhsRes != null){
-                    for(JsonnetObjinside j: rhsRes) output.add(j);
+                if (rhsRes != null) {
+                    output.addAll(Arrays.asList(rhsRes));
                 }
 
             }
 
-            return output.toArray(new JsonnetObjinside[output.size()]);
+            return output.toArray(new JsonnetObjinside[0]);
```

Comment:
@bryant1410 what's going on here? I thought we had to provide an array of the correct size to `toArray` for it to store the data in. How come we can pass in empty arrays?

Reply:
The IDE said:

> Call to `toArray()` with pre-sized array argument 'new JsonnetObjinside[output.size()]' 
_Inspection info_: There are two styles to convert a collection to an array: either using a pre-sized array (like `c.toArray(new String[c.size()])`) or using an empty array (like `c.toArray(new String[0])`).
In older Java versions using a pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However, since late updates of OpenJDK 6, this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the `size` and `toArray` call which may result in extra nulls at the end of the array if the collection was concurrently shrunk during the operation.
This inspection allows following the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).

Reply:
cool, thanks!